File: bidding-engine/src/main/java/com/omnibus/bidding/api/OpenApiConfiguration.java
package com.omnibus.bidding.api;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Contact;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.info.License;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class OpenApiConfiguration {

    @Bean
    public OpenAPI omnibusOpenApi() {
        return new OpenAPI().info(new Info()
            .title("Omnibus Bidding Engine API")
            .description("REST API for bridge bidding recommendations and Drools rule administration.")
            .version("v1")
            .contact(new Contact()
                .name("Omnibus")
                .url("https://example.local/omnibus"))
            .license(new License()
                .name("Internal Use")));
    }
}

File: services/core/rule-engine/src/main/kotlin/pl/writeonly/omnibus/rule/api/OpenApiConfiguration.kt
package pl.writeonly.omnibus.bidding.api

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Contact
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.info.License
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenApiConfiguration {
    @Bean
    fun omnibusOpenApi(): OpenAPI =
        OpenAPI().info(
            Info()
                .title("Omnibus Bidding Engine API")
                .description("REST API for bridge bidding recommendations and Drools rule administration.")
                .version("v1")
                .contact(
                    Contact()
                        .name("Omnibus")
                        .url("https://example.local/omnibus"),
                )
                .license(
                    License()
                        .name("Internal Use"),
                ),
        )
}

